modules/calib3d/src/chessboard.hpp

```diff
@@ -0,0 +1,811 @@
+  If you do not agree to this license, do not download, install,
+  copy or use the software.
+
+
+                          BSD 3-Clause License
+
+ Copyright (C) 2014, Olexa Bilaniuk, Hamid Bazargani & Robert Laganiere, all rights reserved.
+
+ Redistribution and use in source and binary forms, with or without modification,
+ are permitted provided that the following conditions are met:
+
+   * Redistribution's of source code must retain the above copyright notice,
+     this list of conditions and the following disclaimer.
+
+   * Redistribution's in binary form must reproduce the above copyright notice,
+     this list of conditions and the following disclaimer in the documentation
+     and/or other materials provided with the distribution.
+
+   * The name of the copyright holders may not be used to endorse or promote products
+     derived from this software without specific prior written permission.
+
+ This software is provided by the copyright holders and contributors "as is" and
+ any express or implied warranties, including, but not limited to, the implied
+ warranties of merchantability and fitness for a particular purpose are disclaimed.
+ In no event shall the Intel Corporation or contributors be liable for any direct,
+ indirect, incidental, special, exemplary, or consequential damages
+ (including, but not limited to, procurement of substitute goods or services;
+ loss of use, data, or profits; or business interruption) however caused
+ and on any theory of liability, whether in contract, strict liability,
+ or tort (including negligence or otherwise) arising in any way out of
+ the use of this software, even if advised of the possibility of such damage.
+*/
+
+/**
+ * Bilaniuk, Olexa, Hamid Bazargani, and Robert Laganiere. "Fast Target
+ * Recognition on Mobile Devices: Revisiting Gaussian Elimination for the
+ * Estimation of Planar Homographies." In Computer Vision and Pattern
+ * Recognition Workshops (CVPRW), 2014 IEEE Conference on, pp. 119-125.
+ * IEEE, 2014.
+ */
+
+#ifndef CHESSBOARD_HPP_
+#define CHESSBOARD_HPP_
+
+#include "opencv2/core.hpp"
+#include "opencv2/features2d.hpp"
+#include <vector>
+#include <set>
+#include <map>
+
+namespace cv {
+namespace details{
+/**
+ * \brief Fast point sysmetric cross detector based on a localized radon transformation
+ */
+class CV_EXPORTS_W FastX : public cv::Feature2D
+{
+    public:
+        struct CV_EXPORTS_W Parameters
+        {
+            CV_PROP_RW float strength;       //!< minimal strength of a valid junction in dB
+            CV_PROP_RW float resolution;     //!< angle resolution in radians
+            CV_PROP_RW int branches;         //!< the number of branches
+            CV_PROP_RW int min_scale;        //!< scale level [0..8]
+            CV_PROP_RW int max_scale;        //!< scale level [0..8]
+            CV_PROP_RW bool filter;          //!< post filter feature map to improve impulse response
+            CV_PROP_RW bool super_resolution; //!< up-sample
```

Comment:
`CV_EXPORTS_W` / `CV_PROP_RW` / `CV_WRAP` are used for bindings only (like Python API).
This header is internal (not a part of public API), so these macros have no effect.

Reply:
The API was used in combination with https://github.com/D-Alex/rbind to automatically generate a ruby binding compatible with the types wrapped by https://github.com/D-Alex/ropencv. Here, like you said, it makes no sense and I will remove them.